roofit/roostats/src/SPlot.cxx

```diff
@@ -370,47 +422,50 @@ void SPlot::AddSWeight( RooAbsPdf* pdf, const RooArgList &yieldsTmp,
          const RooCmdArg& arg5, const RooCmdArg& arg6, const RooCmdArg& arg7, const RooCmdArg& arg8)
 {
 
-  RooFit::MsgLevel currentLevel =  RooMsgService::instance().globalKillBelow();
-
   // Find Parameters in the PDF to be considered fixed when calculating the SWeights
   // and be sure to NOT include the yields in that list
   RooArgList* constParameters = (RooArgList*)pdf->getParameters(fSData) ;
-  constParameters->remove(yieldsTmp, kTRUE, kTRUE);
+  for (unsigned int i=0; i < constParameters->size(); ++i) {
+    // Need a counting loop since collection is being modified
+    auto& par = (*constParameters)[i];
+    if (std::any_of(yieldsTmp.begin(), yieldsTmp.end(), [&](const RooAbsArg* yield){ return yield->dependsOn(par); })) {
+      constParameters->remove(par, kTRUE, kTRUE);
+      --i;
+    }
+  }
 
 
   // Set these parameters constant and store them so they can later
   // be set to not constant
-  std::vector<RooRealVar*> constVarHolder;
+  std::vector<RooAbsRealLValue*> constVarHolder;
 
   for(Int_t i = 0; i < constParameters->getSize(); i++)
+  {
+    RooAbsRealLValue* varTemp = dynamic_cast<RooAbsRealLValue*>( constParameters->at(i) );
```

Comment:
can these be anything else than `RooAbsRealLValue`s? (i.e. can this be a `static_cast`?)

Reply:
Yes, they can. It's being checked in the constructor. Thanks.